sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,148 @@
+import warnings
+from typing import NamedTuple
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.util.exceptions import SunpyUserWarning
+from sunpy.sun.models import differential_rotation
+
+from sunkit_image.coalignment.decorators import registered_methods
+
+__all__ = ["coalign", "affine_params"]
+
+
+class affine_params(NamedTuple):
+    """
+    A 2x2 scale matrix defining the image scaling along the x and y axes.
+    """
+    scale: tuple[tuple[float, float], tuple[float, float]]
+    """
+    A 2x2 matrix defining the rotation transformation of the image.
+    """
+    rotation_matrix: tuple[tuple[float, float], tuple[float, float]]
+    """
+    A 2-element tuple stores the translation of the image along the x and y
+    axes in pixels.
+    """
+    translation: tuple[float, float]
+
+
+def update_fits_wcs_metadata(reference_map, target_map, affine_params):
+    """
+    Update the metadata of a sunpy Map object based on affine transformation
+    parameters.
+
+    Parameters
+    ----------
+    reference_map : `sunpy.map.Map`
+        The reference map object to which the target map is to be coaligned.
+    target_map : `sunpy.map.Map`
+        The original map object whose metadata is to be updated.
+    affine_params : object
+        An object containing the affine transformation parameters. This object must
+        have attributes for translation (dx, dy), scale, and rotation.
+
+    Returns
+    -------
+    `sunpy.map.Map`
+        A new sunpy map object with updated metadata reflecting the affine transformation.
+    """
+    pc_matrix = target_map.rotation_matrix
+    # Extacting the affine parameters
+    translation = affine_params.translation
+    scale = affine_params.scale
+    rotation_matrix = affine_params.rotation_matrix
+    # Updating the PC matrix
+    new_pc_matrix = pc_matrix @ rotation_matrix
+    reference_coord = reference_map.pixel_to_world(translation[0], translation[1])
```

Comment:
```suggestion
    new_reference_pixel = translation + np.array([target_map.reference_pixel.x.value, target_map.reference_pixel.y.value])
    reference_coord = reference_map.wcs.pixel_to_world(*new_reference_pixel)
```
In cases where there is actually a rotation and a scaling, we will need to generalize this to the actual affine transform.